### PR TITLE
[Doc] Separate FILES() into loading and unloading

### PR DESCRIPTION
**Pull request by EsoragotoSpirit**
Signed-off-by: EsoragotoSpirit <wanglichen@starrocks.com>

## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docs/en/sql-reference/sql-functions/table-functions/files.md [hunk 1]
```diff
@@ -365,9 +344,88 @@ When used with SELECT, FILES() returns the data in the file as a table.
...
+
+See [FILES() for loading - Parameters - data_format](#data_format).
+
+#### StorageCredentialParams
+
+See [FILES() for loading - Parameters - StorageCredentialParams](#storagecredentialparams).
+
+#### unload_data_param
+
+From v3.2 onwards, FILES() supports defining writable files in remote storage for data unloading.
```
Comment (jaogoy) [inline docs/en/sql-reference/sql-functions/table-functions/files.md:412]: duplicate with content in `## FILES() for unloading`

File: docs/en/sql-reference/sql-functions/table-functions/files.md [hunk 2]
```diff
@@ -365,9 +344,88 @@ When used with SELECT, FILES() returns the data in the file as a table.
...
+
+```sql
+-- Supported from v3.2 onwards.
+unload_data_param::=
+    "compression" = "<compression_method>",
```
Comment (jaogoy) [inline docs/en/sql-reference/sql-functions/table-functions/files.md:417]: Directly use { "uncompressed" | "gzip" | "snappy" | "zstd | "lz4" }

File: docs/en/sql-reference/sql-functions/table-functions/files.md [hunk 3]
```diff
@@ -365,9 +344,88 @@ When used with SELECT, FILES() returns the data in the file as a table.
...
+
+See [FILES() for loading - Parameters - StorageCredentialParams](#storagecredentialparams).
+
+#### unload_data_param
+
+From v3.2 onwards, FILES() supports defining writable files in remote storage for data unloading.
+
+```sql
+-- Supported from v3.2 onwards.
+unload_data_param::=
```
Comment (jaogoy) [inline docs/en/sql-reference/sql-functions/table-functions/files.md:416]: add blank space. `unload_data_param ::=`

File: docs/en/sql-reference/sql-functions/table-functions/files.md [hunk 4]
```diff
@@ -365,9 +344,88 @@ When used with SELECT, FILES() returns the data in the file as a table.
...
+| partition_by     | No           | The list of columns that are used to partition data files into different storage paths. Multiple columns are separated by commas (,). FILES() extracts the key/value information of the specified columns and stores the data files under the storage paths featured with the extracted key/value pair. For further instructions, see Example 5. |
+| single           | No           | Whether to unload the data into a single file. Valid values:<ul><li>`true`: The data is stored in a single data file.</li><li>`false` (Default): The data is stored in multiple files if the amount of data unloaded exceeds 512 MB.</li></ul>                  |
+| target_max_file_size | No           | The best-effort maximum size of each file in the batch to be unloaded. Unit: Bytes. Default value: 1073741824 (1 GB). When the size of data to be unloaded exceeds this value, the data will be divided into multiple files, and the size of each file will not significantly exceed this value. Introduced in v3.2.7. |
```
Comment (jaogoy) [inline docs/en/sql-reference/sql-functions/table-functions/files.md:428]: For `examples`, I recommend to add short meaning words for every exmaple.
`Example 1` -> ` Example 1: Query a Parquet file`

Comment (jaogoy) [reply docs/en/sql-reference/sql-functions/table-functions/files.md:428]: And, put `Example 5: unloading data using INSERT INTO FILES` at last.

**Review by wangsimo0 — APPROVED** [review]

**Review by yingtingdong — APPROVED** [review]

**Pull request merged**